src/sentry/scim/endpoints/users.py

```diff
@@ -0,0 +1,154 @@
+            raise PermissionDenied(detail=ERR_ONLY_OWNER)
+        with transaction.atomic():
+            AuthIdentity.objects.filter(
+                user=member.user, auth_provider__organization=organization
+            ).delete()
+            member.delete()
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=member.id,
+                target_user=member.user,
+                event=AuditLogEntryEvent.MEMBER_REMOVE,
+                data=audit_data,
+            )
+
+    def get(self, request, organization, member):
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def patch(self, request, organization, member):
+        for operation in request.data.get("Operations", []):
+            # we only support setting active to False which deletes the orgmember
+            if operation["value"]["active"] is False:
+                self._delete_member(request, organization, member)
+                return Response(status=204)
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def delete(self, request, organization, member):
+        self._delete_member(request, organization, member)
+        return Response(status=204)
+
+
+class OrganizationSCIMUserIndex(SCIMEndpoint):
+    def get(self, request, organization):
+        # note that SCIM doesn't care about changing results as they're queried
+        # TODO: sanitize get parameter inputs?
+        try:
+            filter_val = parse_filter_conditions(request.GET.get("filter"))
+        except Exception:
+            return Response(detail=SCIM_400_INVALID_FILTER, status=400)
+
+        queryset = (
+            OrganizationMember.objects.filter(
+                Q(invite_status=InviteStatus.APPROVED.value),
+                Q(user__is_active=True) | Q(user__isnull=True),
+                organization=organization,
+            )
+            .select_related("user")
+            .order_by("email", "user__email")
+        )
+        if filter_val:
+            queryset = queryset.filter(
+                Q(email__in=filter_val) | Q(user__email__in=filter_val)
+            )  # not including secondary email vals (dups, etc.)
+
+        # TODO: how is queryset ordered?
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
+
+        return self.paginate(
+            request=request,
+            on_results=lambda results: serialize(results, None, OrganizationMemberSCIMSerializer()),
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+            default_per_page=int(request.GET.get("count", 100)),
+            queryset=queryset,
+        )
+
+    def post(self, request, organization):
+        # TODO: confirm mixed case emails get converted to lowercase
```

Comment:
Fun fact: according to the RFCs that define these things, the part of an email address before the `@` should technically be case-sensitive. Downcasing them is safe according to overwhelmingly common usage, however.

Reply:
👍🏼 yup, and it seems like we downcase it in other parts of the product